prototype_source/ios_gpu_workflow.rst

```diff
@@ -0,0 +1,108 @@
+
+.. code:: shell
+
+    cd PYTORCH_ROOT
+    USE_PYTORCH_METAL=ON python setup.py install --cmake
+
+The command above will build a custom pytorch binary from master. The ``install`` argument simply tells ``setup.py`` to override the existing PyTorch on your desktop. Once the build finished, open another terminal to check the PyTorch version to see if the installation was successful. As the time of writing of this recipe, the version is ``1.8.0a0+41237a4``. You might be seeing different numbers depending on when you check out the code from master, but it should be greater than 1.7.0.
+
+.. code:: python
+
+    import torch
+    torch.__version__ #1.8.0a0+41237a4
+
+
+The next step is going to be converting the mobilenetv2 torchscript model to a Metal compatible model. We'll be leveraging the ``optimize_for_mobile`` API from the ``torch.utils`` module. As shown below
+
+.. code:: python
+
+    import torch
+    import torchvision
+    from torch.utils.mobile_optimizer import optimize_for_mobile
+
+    model = torchvision.models.mobilenet_v2(pretrained=True)
+    scripted_model = torch.jit.script(model)
+    optimized_model = optimize_for_mobile(scripted_model, backend='metal')
+    print(torch.jit.export_opnames(optimized_model))
+    torch.jit.save(optimized_model, './mobilenetv2_metal.pt')
+
+Note that the ``torch.jit.export_opnames(optimized_model)`` is going to dump all the optimized operators from the ``optimized_mobile``. If everything works well, you should be able to see the following ops being printed out from the console
+
+
+.. code:: shell
+
+    ['aten::adaptive_avg_pool2d', 
+    'aten::add.Tensor', 
+    'aten::addmm', 
+    'aten::reshape', 
+    'aten::size.int', 
+    'metal::copy_to_host', 
+    'metal_prepack::conv2d_run']
+
+Those are all the ops we need to run the mobilenetv2 model on iOS GPU. Cool! Now that you have the ``mobilenetv2_metal.pt`` saved on your disk, let's move on to the iOS part.
+
+
+Use C++ APIs
+---------------------
+
+In this section, we'll be using the `HelloWorld example <https://github.com/pytorch/ios-demo-app>`_ to demonstrate how to use the C++ APIs. The first thing we need to do is to build a custom LibTorch from Source. Make sure you have deleted the **build** folder from the previous step in PyTorch root directory. Then run the command below
+
+.. code:: shell
+    
+    IOS_ARCH=arm64 USE_PYTORCH_METAL=1 ./scripts/build_ios.sh
+
+Note ``IOS_ARCH`` tells the script to build a arm64 version of Libtorch. This is because in PyTorch, Metal is only available for the iOS devices that support the Apple A9 chip or above. Once the build finished, follow the `Build PyTorch iOS libraries from source <https://pytorch.org/mobile/ios/#build-pytorch-ios-libraries-from-source>`_ section from the iOS tutorial to setup the XCode settings properly. Don't forget to copy the `./mobilenetv2_metal.pt` to your XCode project.
```

Comment:
Please see if the comment in the link below makes sense? https://github.com/pytorch/tutorials/pull/1191/files/77db29b7559ecf29e611e833e34dfb3127fc2ba6#r509812268